src/MultipartStreamBuilder.php

```diff
@@ -104,8 +121,7 @@ public function addResource($name, $resource, array $options = [])
         }
 
         $this->prepareHeaders($name, $stream, $options['filename'], $options['headers']);
-        $this->data[] = ['contents' => $stream, 'headers' => $options['headers'], 'filename' => $options['filename']];
-
+        $this->addData($stream, $options['headers']);
```

Comment:
this means we call createStream one more time in the `addData` method, but because it already is a StreamInterface we can return it immediately? seems fine to me - otherwise it would become much more complicated and that seems not worth it.